src/bsp/src/posix/src/UserInterface.cpp

```diff
@@ -1,14 +1,21 @@
 #include "bsp/UserInterface.h"
 #include <cstdarg>
 #include <cstdio>
+#include <ros/console.h>
 
 int UserInterface::print(const char* format, ...) const {
+    int retValue = 0;
 
-    va_list args;
-    va_start(args, format);
-    int retValue = printf(format, args);
-    printf("\n");
-    va_end(args);
+    // ROS logging methods do not return the standard printf value. So let's just return -1 if there
```

Comment:
Return the number of character printed, use snprintf, it returns the same thing as printf. You can reuse the string  and pass it to ROS_INFO, preventing to format twice.